internal/util/resource/resource.go

```diff
@@ -22,21 +22,25 @@ import (
 	runtimedebug "runtime/debug"
 	"runtime/pprof"
 	"sync"
+	"sync/atomic"
 	"unsafe"
 
 	"github.com/FerretDB/FerretDB/v2/internal/util/devbuild"
 )
 
 // Token should be a field of a tracked object.
 //
-// The underlying type is not struct{} because (from the Go spec)
-// "Two distinct zero-size variables may have the same address in memory",
-// and they do.
-type Token byte
+// Contains a byte field to ensure unique memory addresses, as Go spec states
+// "Two distinct zero-size variables may have the same address in memory".
+// Also stores cleanup handle for proper resource management.
+type Token struct {
+	cleanup atomic.Pointer[runtime.Cleanup]
+	token   byte
```

Comment:
We no longer need that field. The comment should be updated, too.

Reply:
Let me check your suggestions, thanks a lot 👍